sources/models/src/modeled_types/mod.rs

```diff
@@ -62,6 +62,8 @@ pub mod error {
             field: String,
             source: serde_plain::Error,
         },
+        #[snafu(display("Invalid eviction hard '{}': {}", input, msg))]
+        InvalideEvictionHard { input: String, msg: String },
```

Comment:
we can create two different error enums, one for Key and other for value. So, that you don't have to specifically pass a msg string and we can include msg string as part of `#[snafu(display` to create string you want.
Did you refer it from somewhere, where we are using same enum ? I checked `ECSAttributeKey` and `ECSAttributeValue` we are using different error enum there.

Reply:
Your suggestion could be a way as well. The reason why I make them as one is trying to make simple and clear. and I agree with you that different error enums could be a better way. Btw, I got this idea from `KubernetesLableKey` and `KubernetesLabelValue`.

Reply:
The reason why we have error variants like this is because there might be a lot of special cases we check for and capturing each one as a separate error variant would be too much when all we want to convey is the value is invalid for so-and-so reason. An example of this is `InvalidSysctlKey`.

If we only return this error in one location I would use `#[snafu(display)]`. If there's more than one check that might return this, then I think having it like this is fine.